sql: TestListProfilerExecutionDetails failed

sql.TestListProfilerExecutionDetails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19136354?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19136354?buildTab=artifacts#/) on master @ [c5964d7e54c7156e4813183bea5abdaad68f61a1](https://github.com/cockroachdb/cockroach/commits/c5964d7e54c7156e4813183bea5abdaad68f61a1):


```
=== RUN   TestListProfilerExecutionDetails
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestListProfilerExecutionDetails3089870620
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_profiler_execution_details_test.go:440: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestListProfilerExecutionDetails3089870620
--- FAIL: TestListProfilerExecutionDetails (1.23s)
=== RUN   TestListProfilerExecutionDetails/list_execution_detail_files
    jobs_profiler_execution_details_test.go:421: 
        	Error Trace:	pkg/sql/jobs_profiler_execution_details_test.go:421
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/sql/jobs_profiler_execution_details_test.go:415
        	Error:      	"[1/resumer-trace/20250318_182843.39~cockroach.sql.jobs.jobspb.TraceData.binpb 1/resumer-trace/20250318_182843.39~cockroach.sql.jobs.jobspb.TraceData.binpb.txt distsql.20250318_182843.12.html distsql.20250318_182843.52.html goroutines.20250318_182843.19.txt goroutines.20250318_182843.54.txt trace.20250318_182843.20.zip trace.20250318_182843.55.zip]" should have 10 item(s), but has 8
        	Test:       	TestListProfilerExecutionDetails/list_execution_detail_files
--- FAIL: TestListProfilerExecutionDetails/list_execution_detail_files (0.56s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListProfilerExecutionDetails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48649

sql.TestListProfilerExecutionDetails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19137254?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19137254?buildTab=artifacts#/) on master @ [ff8d2bd1b002a1d83b44b07edaa5580290ec9d05](https://github.com/cockroachdb/cockroach/commits/ff8d2bd1b002a1d83b44b07edaa5580290ec9d05):


```
=== RUN   TestListProfilerExecutionDetails
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestListProfilerExecutionDetails3354553608
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_profiler_execution_details_test.go:440: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestListProfilerExecutionDetails3354553608
--- FAIL: TestListProfilerExecutionDetails (0.85s)
=== RUN   TestListProfilerExecutionDetails/list_execution_detail_files
    jobs_profiler_execution_details_test.go:421: 
        	Error Trace:	pkg/sql/jobs_profiler_execution_details_test.go:421
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/sql/jobs_profiler_execution_details_test.go:415
        	Error:      	"[1/resumer-trace/20250318_200810.69~cockroach.sql.jobs.jobspb.TraceData.binpb 1/resumer-trace/20250318_200810.69~cockroach.sql.jobs.jobspb.TraceData.binpb.txt distsql.20250318_200810.49.html distsql.20250318_200810.77.html goroutines.20250318_200810.51.txt goroutines.20250318_200810.78.txt trace.20250318_200810.52.zip trace.20250318_200810.79.zip]" should have 10 item(s), but has 8
        	Test:       	TestListProfilerExecutionDetails/list_execution_detail_files
--- FAIL: TestListProfilerExecutionDetails/list_execution_detail_files (0.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListProfilerExecutionDetails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestListProfilerExecutionDetails [failed](https://tanzanite.cluster.engflow.com/invocations/default/8e205261-14b0-48fb-a183-a437f9c8bc0f?testReportRun=5&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [94db60946a203659fd7a0939439fd1e2fba32b64](https://github.com/cockroachdb/cockroach/commits/94db60946a203659fd7a0939439fd1e2fba32b64):


```
=== RUN   TestListProfilerExecutionDetails
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestListProfilerExecutionDetails2547715028
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_profiler_execution_details_test.go:440: -- test log scope end --
test logs left over in: outputs.zip/logTestListProfilerExecutionDetails2547715028
--- FAIL: TestListProfilerExecutionDetails (0.51s)
=== RUN   TestListProfilerExecutionDetails/list_execution_detail_files
    jobs_profiler_execution_details_test.go:402: 
        	Error Trace:	pkg/sql/jobs_profiler_execution_details_test.go:402
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/sql/jobs_profiler_execution_details_test.go:396
        	Error:      	"[distsql.20250319_055549.34.html goroutines.20250319_055549.36.txt trace.20250319_055549.37.zip]" should have 5 item(s), but has 3
        	Test:       	TestListProfilerExecutionDetails/list_execution_detail_files
--- FAIL: TestListProfilerExecutionDetails/list_execution_detail_files (0.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListProfilerExecutionDetails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestListProfilerExecutionDetails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19143917?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19143917?buildTab=artifacts#/) on master @ [07717228b09021b5a3cea4fd33e4b39dd8e5cf9f](https://github.com/cockroachdb/cockroach/commits/07717228b09021b5a3cea4fd33e4b39dd8e5cf9f):


```
=== RUN   TestListProfilerExecutionDetails
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestListProfilerExecutionDetails4262569852
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_profiler_execution_details_test.go:440: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f4d1e5a362e4384d18c9c26e2defb957/logTestListProfilerExecutionDetails4262569852
--- FAIL: TestListProfilerExecutionDetails (0.81s)
=== RUN   TestListProfilerExecutionDetails/list_execution_detail_files
    jobs_profiler_execution_details_test.go:402: 
        	Error Trace:	pkg/sql/jobs_profiler_execution_details_test.go:402
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/sql/jobs_profiler_execution_details_test.go:396
        	Error:      	"[distsql.20250319_105357.16.html goroutines.20250319_105357.21.txt trace.20250319_105357.22.zip]" should have 5 item(s), but has 3
        	Test:       	TestListProfilerExecutionDetails/list_execution_detail_files
--- FAIL: TestListProfilerExecutionDetails/list_execution_detail_files (0.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListProfilerExecutionDetails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestListProfilerExecutionDetails [failed](https://tanzanite.cluster.engflow.com/invocations/default/f2ba4dd0-059f-4302-b95e-a4a3bf33a8f6?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [3227e31cc49650fae510626b79a2e93614195bc6](https://github.com/cockroachdb/cockroach/commits/3227e31cc49650fae510626b79a2e93614195bc6):


```
=== RUN   TestListProfilerExecutionDetails
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestListProfilerExecutionDetails4284794969
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_profiler_execution_details_test.go:440: -- test log scope end --
test logs left over in: outputs.zip/logTestListProfilerExecutionDetails4284794969
--- FAIL: TestListProfilerExecutionDetails (5.33s)
=== RUN   TestListProfilerExecutionDetails/list_execution_detail_files
    jobs_profiler_execution_details_test.go:421: 
        	Error Trace:	pkg/sql/jobs_profiler_execution_details_test.go:421
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/sql/jobs_profiler_execution_details_test.go:415
        	Error:      	"[1/resumer-trace/20250319_121201.12~cockroach.sql.jobs.jobspb.TraceData.binpb 1/resumer-trace/20250319_121201.12~cockroach.sql.jobs.jobspb.TraceData.binpb.txt distsql.20250319_121200.10.html distsql.20250319_121201.52.html goroutines.20250319_121200.17.txt goroutines.20250319_121201.61.txt trace.20250319_121200.22.zip trace.20250319_121201.66.zip]" should have 10 item(s), but has 8
        	Test:       	TestListProfilerExecutionDetails/list_execution_detail_files
--- FAIL: TestListProfilerExecutionDetails/list_execution_detail_files (2.08s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestListProfilerExecutionDetails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/jobs_profiler_execution_details_test.go
+++ b/pkg/sql/jobs_profiler_execution_details_test.go
@@ -399,7 +399,9 @@ func TestListProfilerExecutionDetails(t *testing.T) {
 			if s.DeploymentMode().IsExternal() {
 				expectedCount--
 			}
-			require.Len(t, files, expectedCount)
+			if len(files) != expectedCount {
+				return errors.Newf("expected %d files, got %d: %v", expectedCount, len(files), files)
+			}
 			return nil
 		})
 
@@ -418,7 +420,9 @@ func TestListProfilerExecutionDetails(t *testing.T) {
 			if s.DeploymentMode().IsExternal() {
 				expectedCount = 8
 			}
-			require.Len(t, files, expectedCount)
+			if len(files) != expectedCount {
+				return errors.Newf("expected %d files, got %d: %v", expectedCount, len(files), files)
+			}
 			return nil
 		})
 		patterns = []string{

